src/BioformatsLoader.jl

```diff
@@ -76,7 +77,36 @@ function interpret_blob!(oxr, arr, sizes)
     return reshape(arr, sizes)
 end
 
-function open_stack(oxr::OMEXMLReader; order="CYXZT")
+function no_colon(sz, subidx)
+    ntuple((d)->typeof(subidx[d]) == Colon ? (1:sz[d]) : 
+            ((typeof(subidx[d]) <: Number) ? (subidx[d]:subidx[d]) : subidx[d]), length(sz))
+end
+
+function sub_indices(sz, subidx)
+    CartesianIndices(no_colon(sz, subidx))
+end
+
+function clipped_size(sz, subidx)
+    return size(sub_indices(sz, subidx))
+end
+
+"""
+    openbytes(oxr, idx, fsubidx, raw_size)
+
+a version of `openbytes` that reads and limits to a set of ranges. `fsubidx` specifies the index range to use and `raw_size` the ND size of this subimage to read and select a region from.
+"""
+function openbytes(oxr, idx, fsubidx, raw_size)
+    arr_nd = openbytes(oxr, idx)
+    arr_nd = reinterpret(getpixeltype(oxr), arr_nd)
+    if all(typeof.(fsubidx) .== Colon)
+        return arr_nd[:]
+    else
+        rv = reshape(arr_nd, raw_size)
+        return (rv[fsubidx...])[:]
```

Comment:
The return value of `openbytes` should be a vector of bytes, the API becomes a little bit confusing if it sometimes returns the pixel type sometimes bytes. This would be solved by using `openBytes(int no, int x, int y, int w, int h)`.